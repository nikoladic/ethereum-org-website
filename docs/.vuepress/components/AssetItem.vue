<template>
  <div class="asset-item pb-1">
    <a :href="imagePath('png')" target="_blank" rel="noopener noreferrer">
      <img class="asset-image pt-1 pb-1 max-w-100" :src="imagePath('png')" />
    </a>
    <div class="l8">
      Download:
      <span v-if="hasFileType('png') === true">
        <a :href="imagePath('png')" target="_blank" rel="noopener noreferrer"
          >PNG</a
        >
      </span>
      <span v-if="hasFileType('svg') === true">
        /
        <a :href="imagePath('svg')" target="_blank" rel="noopener noreferrer"
          >SVG</a
        >
      </span>
      <span v-if="hasFileType('jpg') === true">
        /
        <a :href="imagePath('jpg')" target="_blank" rel="noopener noreferrer"
          >JPG</a
        >
      </span>
    </div>
  </div>
</template>

<script>
export default {
  props: ['assetPath'],
  methods: {
    hasFileType: function(fileType) {
      try {
        require('../public/assets/' + this.assetPath + '.' + fileType)
        return true
      } catch (e) {
        return false
      }
    },
    imagePath: function(fileType) {
      return require('../public/assets/' + this.assetPath + '.' + fileType)
    }
  }
}
</script>

<style lang="stylus" scoped>
@import '../theme/styles/config.styl';

.asset-item
  border-bottom 1px dotted $subduedColor

.asset-image
  max-height 200px
</style>
