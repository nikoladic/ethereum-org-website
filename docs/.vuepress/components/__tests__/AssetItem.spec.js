import { shallowMount } from '@vue/test-utils'
import AssetItem from "@/components/AssetItem"

describe('AssetItem', () => {
  test('render successfully', () => {
    const wrapper = shallowMount(AssetItem, {
      propsData: {
        assetPath: "eth-logo-white-black"
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})

