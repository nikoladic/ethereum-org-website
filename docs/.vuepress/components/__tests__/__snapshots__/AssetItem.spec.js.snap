// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetItem render successfully 1`] = `
<div
  class="asset-item"
>
  <a
    href="[object Object]"
    target="_blank"
  >
    <img
      class="asset-image"
      src="[object Object]"
    />
  </a>
   
  <div
    class="asset-download"
  >
    
    Download:
    
    <span>
      <a
        href="[object Object]"
        target="_blank"
      >
        PNG
      </a>
    </span>
     
    <span>
      
      /
      
      <a
        href="[object Object]"
        target="_blank"
      >
        SVG
      </a>
    </span>
     
    <span>
      
      /
      
      <a
        href="[object Object]"
        target="_blank"
      >
        JPG
      </a>
    </span>
  </div>
</div>
`;
